graphql/exec.go

```diff
@@ -118,3 +123,33 @@ func getOrCreateField(c *[]CollectedField, name string, creator func() Collected
 	*c = append(*c, f)
 	return &(*c)[len(*c)-1]
 }
+
+func shouldIncludeNode(directives ast.DirectiveList, variables map[string]interface{}) bool {
+	skip, include := false, true
+
+	if d := directives.ForName("skip"); d != nil {
+		skip = resolveIfArgument(d, variables)
+	}
+
+	if d := directives.ForName("include"); d != nil {
+		include = resolveIfArgument(d, variables)
+	}
+
+	return !skip && include
+}
+
+func resolveIfArgument(d *ast.Directive, variables map[string]interface{}) bool {
+	arg := d.Arguments.ForName("if")
+	if arg == nil {
+		panic(fmt.Sprintf("%s: argument 'if' not defined", d.Name))
+	}
+	value, err := arg.Value.Value(variables)
+	if err != nil {
+		panic(err)
+	}
+	ret, ok := value.(bool)
+	if !ok {
```

Comment:
@vektah I noticed this panics when a variable is omitted from the query altogether.  "Invalid" values like strings/numbers _seem_ to get coerced into a `bool`. But there's probably somewhere better to report on missing variables?

Reply:
Yeah, theres a post-validation step thats still missing. Ill add a card and make sure its covered before we :shipit: